src/ccl_halomod.c

```diff
@@ -0,0 +1,428 @@
+#include "ccl_utils.h"
+#include "math.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "gsl/gsl_integration.h"
+#include "gsl/gsl_interp.h"
+#include "gsl/gsl_spline.h"
+#include "gsl/gsl_sf_expint.h"
+#include "ccl_placeholder.h"
+#include "ccl_background.h"
+#include "ccl_power.h"
+#include "ccl_massfunc.h"
+#include "ccl_error.h"
+#include "class.h"
+#include "ccl_params.h"
+#include "ccl_emu17.h"
+#include "ccl_emu17_params.h"
+
+// Set two-halo term
+// 1 - Standard two-halo term
+// 2 - Linear theory
+int i2h=1;
```

Comment:
No global variables!!!!

Reply:
Done

Reply:
Although, what is the best thing to do here? There are some variables that I would like to be accessed by multiple functions. It seems laborious to pass them all explicitly between all the routines that might use them, and repeating their definitions in each routine seems like it could lead to errors.